projects/extension/sql/incremental/016-migrate-existing-vectorizers-to-loading.sql

```diff
@@ -0,0 +1,34 @@
+do language plpgsql $block$
+DECLARE
+    _vectorizer RECORD;
+    _chunking jsonb;
+    _chunk_column text;
+    _config jsonb;
+BEGIN
+    -- Loop through all vectorizers
+    FOR _vectorizer IN SELECT id, config FROM ai.vectorizer
+    LOOP
+        -- Extract the chunking config and chunk_column
+        _chunking := _vectorizer.config->'chunking';
```

Comment:
```suggestion
        _chunking := _vectorizer.config operator(pg_catalog.->) 'chunking'; 
```

(applies for the rest of the file)

Based in our codebase, we use operators instead. However, I'm not 100% sure it is mandatory; I agree this implementation looks more readable than with operators. Ping @jgpruitt who can shed some light on this.

Reply:
`->` is considerably more readable.
`operator(pg_catalog.->)` is "safer". It explicitly refers to the `->` operator that exists in the `pg_catalog` schema which is built-in. Only a superuser could change the definition of `->` in `pg_catalog`.
`->` alone resolves the operator based on the current search_path. To execute malicious code, a malicious user could define a new `->` operator in a schema other than `pg_catalog`. This only really comes in to play if the function is `security definer`. A `security definer` function defined in an extension is executed as superuser.